src/hearings/containers/hearing-actuals/hearing-actual-add-edit-summary/hearing-actual-add-edit-summary.component.ts

```diff
@@ -67,4 +75,25 @@ export class HearingActualAddEditSummaryComponent implements OnInit, OnDestroy {
     }
     return '';
   }
+
+  public getHearingResultReasonTypeDescription(hearingOutcome: HearingOutcomeModel): string {
+    if (hearingOutcome.hearingResult === HearingResult.ADJOURNED) {
+      const hearingActualReason = this.adjournHearingActualReasons && this.adjournHearingActualReasons.find(x => x.key === hearingOutcome.hearingResultReasonType);
+      if (hearingActualReason) {
+        return hearingActualReason.value_en;
+      }
+    }
+    if (hearingOutcome.hearingResult === HearingResult.CANCELLED) {
+      const hearingActualReason = this.cancelHearingActualReasons && this.cancelHearingActualReasons.find(x => x.key === hearingOutcome.hearingResultReasonType);
+      if (hearingActualReason) {
+        return hearingActualReason.value_en;
+      }
+    }
```

Comment:
- Seems duplicated code can be extracted to one method
- If x could be more descriptive

Reply:
I have now refactored this method.